pkg/relayer/proxy/proxy.go

```diff
@@ -66,53 +67,61 @@ func NewRelayerProxy(
 	accountQuerier := accounttypes.NewQueryClient(clientCtx)
 	supplierQuerier := suppliertypes.NewQueryClient(clientCtx)
 	sessionQuerier := sessiontypes.NewQueryClient(clientCtx)
-	providedServices := buildProvidedServices(ctx, supplierQuerier)
 	servedRelays, servedRelaysProducer := channel.NewObservable[*types.Relay]()
 
 	return &relayerProxy{
 		// TODO_INCOMPLETE(@red-0ne): Uncomment once the BlockClient interface is available.
-		// blockClient:          blockClient,
+		// blockClient:       blockClient,
 		keyName:              keyName,
 		keyring:              keyring,
 		accountsQuerier:      accountQuerier,
 		supplierQuerier:      supplierQuerier,
 		sessionQuerier:       sessionQuerier,
-		providedServices:     providedServices,
 		servedRelays:         servedRelays,
 		servedRelaysProducer: servedRelaysProducer,
 	}
 }
 
 // Start starts all supported proxies and returns an error if any of them fail to start.
 func (rp *relayerProxy) Start(ctx context.Context) error {
-	panic("TODO: implement relayerProxy.Start")
+	// The provided services map is built from the supplier's on-chain advertised information,
+	// which is a runtime parameter that can be changed by the supplier.
+	// Build the provided services map at Start instead of NewRelayerProxy to avoid having to
+	// return an error from the constructor.
+	err := rp.BuildProvidedServices(ctx)
+	if err != nil {
+		return err
+	}
+
+	eg, gctx := errgroup.WithContext(ctx)
+
+	for _, providedService := range rp.providedServices {
+		for _, svr := range providedService {
```

Comment:
Here's another example of the inconsistency between "server" and "service" nomeclature.